fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java

```diff
@@ -1271,65 +1277,133 @@ private static ColumnDef findPartitionDefByName(List<ColumnDef> columnDefs, Stri
         return null;
     }
 
+    private static class AstVisitorFactory {
+        public static class AstVisitors {
+            public AstVisitor<Expr, Relation> exprShuttle;
```

Comment:
Why do need to keep these two fields in this class?

Reply:
`exprShuttle` and `slotRefResolver` will reference each other, so a factory class is needed to encapsulate them